link_release.sh

```diff
@@ -2,76 +2,74 @@
 TEMPLATE_FILE="template.json"
 
 echo "Generating new $VERSION release for $NAME"
-echo "debug"
 
-# can simplify this into a for loop...
-if test -f "${DL_LINUX}";
-then 
-  sha_linux=$(shasum -a 256 $DL_LINUX | cut -d ' ' -f 1)
-  echo $sha_linux
-fi
+DLS=("$DL_LINUX" "$DL_MAC" "$DL_WIN")
+DIR=("$DIR_LINUX" "$DIR_MAC" "$DIR_WIN")
 
-if test -f "${DL_MAC}";
-then 
-  sha_macos=$(shasum -a 256 $DL_MAC | cut -d ' ' -f 1)
-  echo $sha_macos
-fi
+DL_LINUX_ARCH=("$DL_LINUX_X86_64" "$DL_LINUX_AARCH64" "$DL_LINUX_ARM")
+DL_MAC_ARCH=("$DL_MAC_X86_64" "$DL_MAC_AARCH64" "$DL_MAC_ARM")
+DL_WIN_ARCH=("$DL_WIN_X86_64" "$DL_WIN_AARCH64" "$DL_WIN_ARM")
+
+# shellcheck disable=SC2016
+arch_specific='{"x86_64":$x86_64,"aarch64":$aarch,"arm":$arm}'
+
+# Compute all architecture structs and shas
+for ((i=0; i<3; i++)); do # iterate over platforms, linux, mac, win
```

Comment:
this is getting out of hand

Reply:
maybe should just move to rust instead of trying to do this all in shell (downside being having to install rust when generating json)